elasticapm/traces.py

```diff
@@ -1266,7 +1266,34 @@ def set_context(data, key="custom"):
         transaction.context[key] = data
 
 
+def set_span_context(data: Union[dict, Callable], key: str = "custom") -> None:
+    """
+    Attach contextual data to the current span.
+
+    If the parent transaction is not sampled, this function becomes a no-op.
+
+    :param data: a dictionary, or a callable that returns a dictionary
+    :param key: the namespace for this data
+    """
+    span = execution_context.get_span()
+    if isinstance(span, DroppedSpan):
+        return
+    if callable(data):
+        data = data()
+
+    # remove invalid characters from key names
+    for k in list(data.keys()):
+        if LABEL_RE.search(k):
+            data[LABEL_RE.sub("_", k)] = data.pop(k)
+
+    if key in span.context:
+        span.context[key].update(data)
+    else:
+        span.context[key] = data
+
+
 set_custom_context = functools.partial(set_context, key="custom")
+set_custom_span_context = functools.partial(set_span_context, key="custom")
```

Comment:
The intake API schema (span.json) doesn't include `context.custom`, so will these be ingested, or just dropped?

Reply:
Good question! I will do some investigation.

Reply:
You're correct; the data is dropped. I think rather than spending the time on apm-data getting this added, I'm going to recommend labels to the user.